### PR TITLE
Remove use_scipy argument in rotate functions

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes https://github.com/sunpy/sunpy/issues/6530

### CODE REVIEW

File: sunpy/image/transform.py [hunk 1]
```diff
@@ -90,35 +90,19 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
...
+    method = _get_transform_method(method)
 
     # Transform the image using the appropriate function
     rotated_image = _rotation_registry[method].function(image, rmatrix, shift, order, missing, clip)
 
     return rotated_image
 
 
-def _get_transform_method(method, use_scipy):
+def _get_transform_method(method):
```
Comment (nabobalis) [inline sunpy/image/transform.py:101]: I didn't really want to remove this function as it was more "work" but as it stands it doesn't do anything that can't be copy and pasted into each method.

**Review by dstansby — APPROVED** [review]

File: changelog/6538.removal.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+:meth:`sunpy.image.transform.affine_transform` and :func:`sunpy.map.GenericMap.rotate` will no longer accept the ``use_scipy`` keyword.
```
Comment (wtbarnes) [inline changelog/6538.removal.rst:1]:
```suggestion
:meth:`sunpy.image.transform.affine_transform` and :func:`sunpy.map.GenericMap.rotate` no longer accepts the ``use_scipy`` keyword.
```

File: sunpy/image/transform.py [hunk 3]
```diff
@@ -90,35 +90,19 @@ def affine_transform(image, rmatrix, order=3, scale=1.0, image_center=None,
...
-        if use_scipy is True and method != 'scipy':
-            warn_user(f"Using scipy instead of {method} for rotation.")
-            method = 'scipy'
-
-    if method == 'scikit-image':
-        try:
-            import skimage  # NoQA
-        except ImportError:
-            raise ImportError("scikit-image must be installed to be usable for rotation.")
-
```
Comment (wtbarnes) [inline sunpy/image/transform.py:121]: Why move this block out of here? Isn't the point of the `_get_method` function to do validation of the choice of method all in one place?

File: sunpy/image/transform.py [hunk 4]
```diff
@@ -336,7 +320,10 @@ def _rotation_skimage(image, matrix, shift, order, missing, clip):
     * Does not pass NaN as ``missing`` to :func:`~skimage.transform.warp` due to
       inconsistent handling across interpolation orders
     """
-    import skimage.transform
+    try:
+        import skimage.transform
+    except ImportError:
+        raise ImportError("The scikit-image package is required to use this rotation method.")
```
Comment (wtbarnes) [inline sunpy/image/transform.py:326]: See comment above

**Review by wtbarnes — APPROVED** [review]
Just a few comments, but nothing to block on. I'm just a bit confused by the reorganization of things.

File: sunpy/image/transform.py (hunk 3 above)
Comment (nabobalis) [reply sunpy/image/transform.py:121]: The rotation method for opencv check for the import inside of them and I think they should handle it.

The same goes for cupy, if it was done here, someone would need to PR it to core sunpy.

Comment (wtbarnes) [reply sunpy/image/transform.py:121]: Ah ok, thanks. That makes sense. So `_get_method` should just check the registry and return the desired method.

Comment (nabobalis) [reply sunpy/image/transform.py:121]: That is what it does now only, before it also did the import check only for scikit image.

**Pull request merged**